contracts/SimpleOperations.sol

```diff
@@ -45,121 +46,258 @@ contract SimpleOperations is Operations {
+		);
 	}
 
 	// Admin functions
 
-	function addClient(bytes32 _client, address _owner) public only_owner {
-		client[_client].owner = _owner;
-		clientOwner[_owner] = _client;
-		ClientAdded(_client, _owner);
-	}
+	function setClient(bytes32 _client, address _owner)
+		public
+		onlyOwner
+		notClientOwner(_owner)
+	{
+		address owner = client[_client].owner;
+		delete clientOwner[owner];
 
-	function removeClient(bytes32 _client) public only_owner {
-		resetClientOwner(_client, 0);
-		delete client[_client];
-		ClientRemoved(_client);
+		clientOwner[_owner] = _client;
+		client[_client].owner = _owner;
 	}
 
-	function resetClientOwner(bytes32 _client, address _newOwner) public only_owner {
-		var old = client[_client].owner;
-		ClientOwnerChanged(_client, old, _newOwner);
-		clientOwner[old] = bytes32(0);
-		clientOwner[_newOwner] = _client;
-		client[_client].owner = _newOwner;
+	// Removes the client. This only removes the ownership of the client from the current client
+	// owner, the existing release and build data is kept. The getters are guarded against this
+	// behavior, and check that the client is currently not removed (i.e. no owner) before accessing
+	// the data. If a removed client is then re-added (with `setClient`) all of the previous release
+	// and build information is available and returned by the getters.
+	function removeClient(bytes32 _client)
+		public
+		onlyOwner
+	{
+		address owner = client[_client].owner;
+		delete clientOwner[owner];
+		delete client[_client].owner;
 	}
 
-	function setLatestFork(uint32 _forkNumber) public only_owner {
-		ForkRatified(_forkNumber);
+	function setLatestFork(uint32 _forkNumber)
+		public
+		onlyOwner
+	{
+		emit ForkRatified(_forkNumber);
 		latestFork = _forkNumber;
 	}
 
-	function setOwner(address _newOwner) public only_owner {
-		OwnerChanged(grandOwner, _newOwner);
+	function setOwner(address _newOwner)
+		public
+		onlyOwner
+	{
+		emit OwnerChanged(grandOwner, _newOwner);
 		grandOwner = _newOwner;
 	}
 
 	// Getters
 
-	function isLatest(bytes32 _client, bytes32 _release) public view returns (bool) {
-		return latestInTrack(_client, track(_client, _release)) == _release;
+	function isLatest(bytes32 _client, bytes32 _release)
+		public
+		view
+		returns (bool)
+	{
+		return clientExists(_client) && latestInTrack(_client, track(_client, _release)) == _release;
 	}
 
-	function track(bytes32 _client, bytes32 _release) public view returns (uint8) {
-		return client[_client].release[_release].track;
+	function track(bytes32 _client, bytes32 _release)
+		public
+		view
+		returns (uint8)
+	{
+		if (clientExists(_client)) {
+			return client[_client].release[_release].track;
+		} else {
+			return 0;
+		}
 	}
 
-	function latestInTrack(bytes32 _client, uint8 _track) public view returns (bytes32) {
-		return client[_client].current[_track];
+	function latestInTrack(bytes32 _client, uint8 _track)
+		public
+		view
+		returns (bytes32)
+	{
+		if (clientExists(_client)) {
```

Comment:
Why not just `require(clientExists(_client))`?

Reply:
I thought it would be better to keep the existing interface (similar to returning `None`) rather than abort.

Reply:
ok, it's fine then.